File: binding/src/libqpolygon.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpolygon.html
pub const qpolygon = struct {

    /// New constructs a new QPolygon object.
    pub fn New() QtC.QPolygon {
        return qtc.QPolygon_new();
    }


    /// New2 constructs a new QPolygon object.
    pub fn New2(v: []const u8) QtC.QPolygon {
        return qtc.QPolygon_new2(@ptrCast(v));
    }


    /// New3 constructs a new QPolygon object.
    pub fn New3(r: ?*anyopaque) QtC.QPolygon {
        return qtc.QPolygon_new3(@ptrCast(r));
    }


    /// New4 constructs a new QPolygon object.
    pub fn New4(nPoints: i32, points: *const i32) QtC.QPolygon {
        return qtc.QPolygon_new4(nPoints, @ptrCast(points));
    }


    /// New5 constructs a new QPolygon object.
    pub fn New5(r: ?*anyopaque, closed: bool) QtC.QPolygon {
        return qtc.QPolygon_new5(@ptrCast(r), closed);
    }


    pub fn OperatorQVariant(self: ?*anyopaque, ) QtC.QVariant {
        return qtc.QPolygon_OperatorQVariant(@ptrCast(self));
    }

    pub fn Translate(self: ?*anyopaque, dx: i32, dy: i32) void {
        qtc.QPolygon_Translate(@ptrCast(self), dx, dy);
    }

    pub fn Translate2(self: ?*anyopaque, offset: ?*anyopaque) void {
        qtc.QPolygon_Translate2(@ptrCast(self), @ptrCast(offset));
    }

    pub fn BoundingRect(self: ?*anyopaque, ) QtC.QRect {
        return qtc.QPolygon_BoundingRect(@ptrCast(self));
    }

    pub fn Point(self: ?*anyopaque, i: i32, x: *i32, y: *i32) void {
        qtc.QPolygon_Point(@ptrCast(self), i, @ptrCast(x), @ptrCast(y));
    }

    pub fn Point2(self: ?*anyopaque, i: i32) QtC.QPoint {
        return qtc.QPolygon_Point2(@ptrCast(self), i);
    }

    pub fn SetPoint(self: ?*anyopaque, index: i32, x: i32, y: i32) void {
        qtc.QPolygon_SetPoint(@ptrCast(self), index, x, y);
    }

    pub fn SetPoint2(self: ?*anyopaque, index: i32, p: ?*anyopaque) void {
        qtc.QPolygon_SetPoint2(@ptrCast(self), index, @ptrCast(p));
    }

    pub fn SetPoints(self: ?*anyopaque, nPoints: i32, points: *const i32) void {
        qtc.QPolygon_SetPoints(@ptrCast(self), nPoints, @ptrCast(points));
    }

    pub fn SetPoints2(self: ?*anyopaque, nPoints: i32, firstx: i32, firsty: i32, : ...) void {
        qtc.QPolygon_SetPoints2(@ptrCast(self), nPoints, firstx, firsty, );
    }

    pub fn PutPoints(self: ?*anyopaque, index: i32, nPoints: i32, points: *const i32) void {
        qtc.QPolygon_PutPoints(@ptrCast(self), index, nPoints, @ptrCast(points));
    }

    pub fn PutPoints2(self: ?*anyopaque, index: i32, nPoints: i32, firstx: i32, firsty: i32, : ...) void {
        qtc.QPolygon_PutPoints2(@ptrCast(self), index, nPoints, firstx, firsty, );
    }

    pub fn ContainsPoint(self: ?*anyopaque, pt: ?*anyopaque, fillRule: i32) bool {
        return qtc.QPolygon_ContainsPoint(@ptrCast(self), @ptrCast(pt), @intCast(fillRule));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPolygon_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qpolygonf.html
pub const qpolygonf = struct {

    /// New constructs a new QPolygonF object.
    pub fn New() QtC.QPolygonF {
        return qtc.QPolygonF_new();
    }


    /// New2 constructs a new QPolygonF object.
    pub fn New2(v: []const u8) QtC.QPolygonF {
        return qtc.QPolygonF_new2(@ptrCast(v));
    }


    /// New3 constructs a new QPolygonF object.
    pub fn New3(r: ?*anyopaque) QtC.QPolygonF {
        return qtc.QPolygonF_new3(@ptrCast(r));
    }


    pub fn OperatorQVariant(self: ?*anyopaque, ) QtC.QVariant {
        return qtc.QPolygonF_OperatorQVariant(@ptrCast(self));
    }

    pub fn Translate(self: ?*anyopaque, dx: f64, dy: f64) void {
        qtc.QPolygonF_Translate(@ptrCast(self), dx, dy);
    }

    pub fn Translate2(self: ?*anyopaque, offset: ?*anyopaque) void {
        qtc.QPolygonF_Translate2(@ptrCast(self), @ptrCast(offset));
    }

    pub fn IsClosed(self: ?*anyopaque, ) bool {
        return qtc.QPolygonF_IsClosed(@ptrCast(self));
    }

    pub fn BoundingRect(self: ?*anyopaque, ) QtC.QRectF {
        return qtc.QPolygonF_BoundingRect(@ptrCast(self));
    }

    pub fn ContainsPoint(self: ?*anyopaque, pt: ?*anyopaque, fillRule: i32) bool {
        return qtc.QPolygonF_ContainsPoint(@ptrCast(self), @ptrCast(pt), @intCast(fillRule));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPolygonF_Delete(@ptrCast(self));
    }
};

File: binding/src/libqpolygon.cpp
#include <QPoint>
#include <QPointF>
#include <QRect>
#include <QRectF>
#include <QVariant>
#include <qpolygon.h>
#include "libqpolygon.h"
#include "libqpolygon.hxx"

QPolygon* QPolygon_new() {
	 return new QPolygon();
}

QPolygon* QPolygon_new2(const libqt_list v) {
	 return new QPolygon(QList<QPoint>());
}

QPolygon* QPolygon_new3(const QRect* r) {
	 return new QPolygon(*r);
}

QPolygon* QPolygon_new4(int nPoints, const int* points) {
	 return new QPolygon(nPoints, points);
}

QPolygon* QPolygon_new5(const QRect* r, bool closed) {
	 return new QPolygon(*r, closed);
}

QVariant* QPolygon_OperatorQVariant(const QPolygon* self) {
	return new QVariant(self->operator QVariant());
}

void QPolygon_Translate(QPolygon* self, int dx, int dy) {
	self->translate(dx, dy);
}

void QPolygon_Translate2(QPolygon* self, const QPoint* offset) {
	self->translate(*offset);
}

QRect* QPolygon_BoundingRect(const QPolygon* self) {
	return new QRect(self->boundingRect());
}

void QPolygon_Point(const QPolygon* self, int i, int* x, int* y) {
	self->point(i, x, y);
}

QPoint* QPolygon_Point2(const QPolygon* self, int i) {
	return new QPoint(self->point(i));
}

void QPolygon_SetPoint(QPolygon* self, int index, int x, int y) {
	self->setPoint(index, x, y);
}

void QPolygon_SetPoint2(QPolygon* self, int index, const QPoint* p) {
	self->setPoint(index, *p);
}

void QPolygon_SetPoints(QPolygon* self, int nPoints, const int* points) {
	self->setPoints(nPoints, points);
}

void QPolygon_SetPoints2(QPolygon* self, int nPoints, int firstx, int firsty, ... ) {
	self->setPoints(nPoints, firstx, firsty, );
}

void QPolygon_PutPoints(QPolygon* self, int index, int nPoints, const int* points) {
	self->putPoints(index, nPoints, points);
}

void QPolygon_PutPoints2(QPolygon* self, int index, int nPoints, int firstx, int firsty, ... ) {
	self->putPoints(index, nPoints, firstx, firsty, );
}

bool QPolygon_ContainsPoint(const QPolygon* self, const QPoint* pt, int fillRule) {
	return self->containsPoint(*pt, static_cast<Qt::FillRule>(fillRule));
}

void QPolygon_Delete(QPolygon* self) {
    delete self;
}

QPolygonF* QPolygonF_new() {
	 return new QPolygonF();
}

QPolygonF* QPolygonF_new2(const libqt_list v) {
	 return new QPolygonF(QList<QPointF>());
}

QPolygonF* QPolygonF_new3(const QRectF* r) {
	 return new QPolygonF(*r);
}

QVariant* QPolygonF_OperatorQVariant(const QPolygonF* self) {
	return new QVariant(self->operator QVariant());
}

void QPolygonF_Translate(QPolygonF* self, double dx, double dy) {
	self->translate(dx, dy);
}

void QPolygonF_Translate2(QPolygonF* self, const QPointF* offset) {
	self->translate(*offset);
}

bool QPolygonF_IsClosed(const QPolygonF* self) {
	return self->isClosed();
}

QRectF* QPolygonF_BoundingRect(const QPolygonF* self) {
	return new QRectF(self->boundingRect());
}

bool QPolygonF_ContainsPoint(const QPolygonF* self, const QPointF* pt, int fillRule) {
	return self->containsPoint(*pt, static_cast<Qt::FillRule>(fillRule));
}

void QPolygonF_Delete(QPolygonF* self) {
    delete self;
}

